### PR TITLE
Update draftable edition sections

### PR DESCRIPTION
**Pull request by Ayora29**
- FileSection.java : Use a LinkedHashMap to parse edition section, preserving section order and improve performance.
- CardEdition.java : Increment art index when multiple cards with the same name are in the same sheet. Add new section names, remove irrelevant ones.
- Edition txt files of draftable sets : update "sections" and "Booster=" parts to better reflect reallity (cf scryfall and mtg.fandom.com) and reduce amount of unecessary additionnal sections/sheets.
- printsheets.txt : remove unecessary or dead sections
- No card addition, no card remove, no cards in different order compared to before (exept some basic lands)
- I have tested drafts for all editions that I have modified

### CODE REVIEW

File: forge-gui/res/editions/30th Anniversary Edition.txt [hunk 1]
```diff
@@ -4,7 +4,7 @@ Date=2022-11-28
 Name=30th Anniversary Edition
 Type=Collector_Edition
 BoosterCovers=3
-Booster=7 Common:fromsheet("30A cards"), 3 Uncommon:fromSheet("30A cards"), 1 Rare:fromSheet("30A cards"), 2 BasicLand:fromSheet("30A cards"), 1 fromSheet("30A retrolands"), 1 fromSheet("30A retrocards")
+Booster=7 Common:fromsheet("30A cards"), 3 Uncommon:fromSheet("30A cards"), 1 Rare:fromSheet("30A cards"), 2 BasicLand:fromSheet("30A cards"), 1 BasicLand:fromSheet("30A retro frame"), 1 !BasicLand:fromSheet("30A retro frame")
```
Comment (tool4ever) [inline forge-gui/res/editions/30th Anniversary Edition.txt:7]: why is the last slot now nonBasicLand? was this an error?

Comment (Ayora29) [reply forge-gui/res/editions/30th Anniversary Edition.txt:7]: Before, there was a section "retrolands" with all the retro frame basic land and a section "retrocards" with all the retro frame non basic land.
I have just simplified using 1 BasicLand:fromSheet("30A retro frame") and 1 !BasicLand:fromSheet("30A retro frame")

Comment (Ayora29) [reply forge-gui/res/editions/30th Anniversary Edition.txt:7]: Now that correct art index is set in CardEdition.java#512, there is no need to additional section that contains the art index. Indicating "1 BasicLand:fromSheet("30A retro frame")" will take the correct art index.

Comment (tool4ever) [reply forge-gui/res/editions/30th Anniversary Edition.txt:7]: ah I understand
I'm not sure if that slot is really limited to nonBasic but this isn't my best area anyway :)

Comment (Ayora29) [reply forge-gui/res/editions/30th Anniversary Edition.txt:7]: There was only basic lands in the former section "30A retrolands", so yes it is only limited to basic lands :)

File: forge-gui/res/editions/Alchemy Horizons Baldur's Gate.txt [hunk 2]
```diff
@@ -93,7 +84,7 @@ ScryfallCode=HBG
 72 R Jon Irenicus, the Exile @Igor Grechanyi
 73 U Liara of the Flaming Fist @David Rapoza
 74 U Minthara of the Absolute @Evyn Fong
-75 M Tasha, Unholy Archmage @Martina Fackova
+75 M Tasha, Unholy Archmage @Martina Fačková
```
Comment (tehdiplomat) [inline forge-gui/res/editions/Alchemy Horizons Baldur's Gate.txt:87]: Forge has had issues historically with these types of characters. I wouldn't add these without careful consideration

File: forge-gui/res/editions/Battlebond.txt [hunk 3]
```diff
@@ -262,9 +262,81 @@ ScryfallCode=BBD
 252 L Swamp @Titus Lunter
 253 L Mountain @Titus Lunter
 254 L Forest @Titus Lunter
+
+[etched]
 255 M Will Kenrith @Anna Steinbauer
 256 M Rowan Kenrith @Anna Steinbauer
 
+[RareMythic]
```
Comment (tehdiplomat) [inline forge-gui/res/editions/Battlebond.txt:270]: I don't really like the idea of using RareMythic as a name here, since it's a keyword for editions. (Which is probably why the previous version was using it incorrectly)

File: forge-gui/res/editions/Conspiracy Take the Crown.txt [hunk 4]
```diff
@@ -5,7 +5,7 @@ Name=Conspiracy: Take the Crown
 Code2=CN2
 Type=Draft
 BoosterCovers=3
-Booster=10 Common:!fromSheet("CN2 Draft Matters"), 3 Uncommon:!fromSheet("CN2 Draft Matters"), 1 RareMythic:!fromSheet("CN2 Draft Matters"), 1 fromSheet("CN2 Draft Matters")
+Booster=10 Common:!fromSheet("CN2 Draft Matters"), 3 Uncommon:!fromSheet("CN2 Draft Matters"), 1 RareMythic:!fromSheet("CN2 Draft Matters"):fromSheet("CN2 cards"), 1 fromSheet("CN2 Draft Matters")
```
Comment (tehdiplomat) [inline forge-gui/res/editions/Conspiracy Take the Crown.txt:8]: I Don't understand this change?

**Review by tehdiplomat — COMMENTED** [review]
I still feel like the default for a booster slot should be "SETCODE Cards" and we shouldn't have to explicitly say we're using them.

File: forge-gui/res/editions/Masters Edition IV.txt [hunk 5]
```diff
@@ -5,7 +5,7 @@ Name=Masters Edition IV
 Code2=ME4
 Type=Online
 BoosterCovers=1
-Booster=10 Common, 3 Uncommon, 1 Rare, 1 fromSheet("ME4 UrzaLands")
+Booster=10 Common, 3 Uncommon, 1 Rare, 1 BasicLand
```
[the remark this answers is not in the document]
Comment (Ayora29) [reply forge-gui/res/editions/Masters Edition IV.txt:8]: They were already flaged with "L" rarity in the file, I didn't modify that.
{3C3395A3-72E9-4FA7-91FF-457EB7C4BA90}
As you said, in this set, Urza lands are considered as basic lands. The "1 BasicLand" here will catch 1 random Urza land. Tested in draft.

File: forge-gui/res/editions/Bloomburrow.txt [hunk 6]
```diff
@@ -21,21 +22,16 @@ Base=Uncommon:fromSheet("BLB cards")
...
 
 [FullArtLand]
-Base=Land:fromSheet("BLB full art")
-Replace=.20F Land:fromSheet("BLB full art")+
+Base=BasicLand:fromSheet("BLB cards")
+Replace=.20F BasicLand:fromSheet("BLB cards")+
 
 [Wildcard]
 Base=Common:fromSheet("BLB cards")
 # I don't know about these numbers, so i'm just copying them from MH3
```
[the remark this answers is not in the document]
Comment (Ayora29) [reply forge-gui/res/editions/Bloomburrow.txt]: I didn't find the exact numbers for rarity distribution of the wildcard slot for BLB. So I have kept this comment.

File: forge-gui/res/editions/Alchemy Horizons Baldur's Gate.txt (hunk 2 above)
Comment (Ayora29) [reply forge-gui/res/editions/Alchemy Horizons Baldur's Gate.txt:87]: Ok, I saw other Artist name with accent/diacritic so I thought is was ok. But if you tell me its not, I can remove it, no problem.

File: forge-gui/res/editions/Conspiracy Take the Crown.txt (hunk 4 above)
Comment (Ayora29) [reply forge-gui/res/editions/Conspiracy Take the Crown.txt:8]: It is only to not catch the promo card 
{7C2770E5-F756-4CC2-92F6-C65489EC32CB}
Otherwise Kaya as 100% more chance to shows up in a pack I guess.

File: forge-gui/res/editions/Bloomburrow.txt (hunk 6 above)
[the remark this answers is not in the document]
Comment (Ayora29) [reply forge-gui/res/editions/Bloomburrow.txt]: Or maybe should I simply use :
[Wildcard]
Any:fromSheet("BLB cards")

File: forge-gui/res/editions/Murders at Karlov Manor.txt [hunk 7]
```diff
@@ -4,7 +4,7 @@ Date=2024-02-09
 Name=Murders at Karlov Manor
 Type=Expansion
 ScryfallCode=MKM
-Booster=7 Common, 3 Uncommon, 1 RareMythic, 1 BasicLand, 1 Any, 1 Any+
+Booster=7 Common:fromSheet("MKM cards"), 3 Uncommon:fromSheet("MKM cards"), 1 RareMythic:fromSheet("MKM cards"), 1 BasicLand, 1 Any:fromSheet("MKM cards"), 1 Any:fromSheet("MKM cards")+
```
[the remark this answers is not in the document]
Comment (Ayora29) [reply forge-gui/res/editions/Murders at Karlov Manor.txt]: Done!

File: forge-gui/res/editions/Battlebond.txt (hunk 3 above)
Comment (Ayora29) [reply forge-gui/res/editions/Battlebond.txt]: Ok. renamed.

File: forge-gui/res/editions/Alchemy Horizons Baldur's Gate.txt (hunk 2 above)
Comment (paulsnoops) [reply forge-gui/res/editions/Alchemy Horizons Baldur's Gate.txt:87]: There was a change at some point that supported this

Comment (Ayora29) [reply forge-gui/res/editions/Alchemy Horizons Baldur's Gate.txt:87]: Indeed, "Martina Fačková" is present in several other edition file with those characters

File: forge-gui/res/editions/Alchemy Horizons Baldur's Gate.txt [hunk 8]
```diff
@@ -337,3 +328,32 @@ A239 U A-Krydle of Baldur's Gate @Bryan Sola
 A243 M A-Minsc & Boo, Timeless Heroes @Andreas Zafiratos
 A259 C A-Lantern of Revealing @Eytan Zana
 A262 U A-Navigation Orb @Robin Olausson
+
+[conjured]
+926 R Lightning Bolt @Irina Nordsol
+927 C Naturalize @Tim Hildebrandt
+928 C Plummet @Alix Branwyn
+45a R Hag of Ceaseless Torment @Konstantin Porubov
```
Comment (paulsnoops) [inline forge-gui/res/editions/Alchemy Horizons Baldur's Gate.txt:336]: The hags are unique cards that don't exist anywhere else, do the images still work correctly?

File: forge-gui/res/editions/Alchemy Horizons Baldur's Gate.txt [hunk 9]
```diff
@@ -337,3 +328,32 @@ A239 U A-Krydle of Baldur's Gate @Bryan Sola
 A243 M A-Minsc & Boo, Timeless Heroes @Andreas Zafiratos
 A259 C A-Lantern of Revealing @Eytan Zana
 A262 U A-Navigation Orb @Robin Olausson
+
+[conjured]
```
Comment (paulsnoops) [inline forge-gui/res/editions/Alchemy Horizons Baldur's Gate.txt:332]: Does this "add" new cards to the edition that will require images? We intentionally left out conjured cards for this reason

Comment (Ayora29) [reply forge-gui/res/editions/Alchemy Horizons Baldur's Gate.txt:332]: Those cards where in the [cards] section before my update so I guess they were already requiring images didn't they?
I have put them under a new section to exclude them from Booster generation, that's all.

File: forge-gui/res/editions/Alchemy Horizons Baldur's Gate.txt (hunk 8 above)
Comment (Ayora29) [reply forge-gui/res/editions/Alchemy Horizons Baldur's Gate.txt:336]: Yes, I have tested to play "The Hourglass Coven" card, and all the "hag" I have conjured display correctly.
I have added the "conjured" section in the known section in file CardEdition.java, so they are loaded correctly

File: forge-gui/res/editions/Alchemy Horizons Baldur's Gate.txt (hunk 9 above)
Comment (Ayora29) [reply forge-gui/res/editions/Alchemy Horizons Baldur's Gate.txt:332]: Oh! they weren't in [cards] actually, I mixed up with the hag ones.

Comment (paulsnoops) [reply forge-gui/res/editions/Alchemy Horizons Baldur's Gate.txt:332]: The hags were in the edition yes but the rest of conjured were not. We typically don't include conjured because they "add" another version of the card that don't even have unique art and therefore cause bloat. Conjure by default will pull a version of that card from another set.

Comment (Ayora29) [reply forge-gui/res/editions/Alchemy Horizons Baldur's Gate.txt:332]: In fact they seems to have unique art (for some of them at least).
https://www.cardmarket.com/en/Magic/Products/Singles/Commander-Legends-Battle-for-Baldurs-Gate/Lightning-Bolt

Comment (Ayora29) [reply forge-gui/res/editions/Alchemy Horizons Baldur's Gate.txt:332]: Is the version of that card from another set is from a random set?

Comment (paulsnoops) [reply forge-gui/res/editions/Alchemy Horizons Baldur's Gate.txt:332]: That art is in CLB though.. https://scryfall.com/card/clb/187/lightning-bolt

I'd rather we stay away from adding conjured cards to editions

Comment (Ayora29) [reply forge-gui/res/editions/Alchemy Horizons Baldur's Gate.txt:332]: Yes sure, I understand, will remove them, apart from the Hag ones

Comment (Ayora29) [reply forge-gui/res/editions/Alchemy Horizons Baldur's Gate.txt:332]: Ok done, I have removed them.

File: forge-gui/res/editions/Portal.txt [hunk 10]
```diff
@@ -16,7 +16,6 @@ ScryfallCode=POR
 4 U Ardent Militia @Mike Raabe
 5 R Armageddon @John Avon
 6 C Armored Pegasus @Andrew Robinson
-6d C Armored Pegasus @Andrew Robinson
```
Comment (tehdiplomat) [inline forge-gui/res/editions/Portal.txt:19]: So why were these removed? These are different arts: 
> Demo Game Armored Pegasus
The demo pack versions of some Portal cards had extra reminder text.

**Comment by tehdiplomat** [comment]
I think we'll merge this after the Foundations release. Thanks for cleaning all this up.

**Comment by Ayora29** [comment]
Ok great :)

**Review by paulsnoops — APPROVED** [review]

**Comment by Agetian** [comment]
Is this MR ready for merge?

**Comment by Agetian** [comment]
@Ayora29 Sadly, had to revert the PR for now due to a crash on startup. Can you please investigate and resubmit the PR after the fix? Thank you in advance!

**Comment by Ayora29** [comment]
@Agetian Ok strange. Will have a look at it. thanks.

**Comment by Ayora29** [comment]
@Agetian There was a conflict with recent update on MH3 edition file. I have fixed it and re-submit PR https://github.com/Card-Forge/forge/pull/6631
Tested.
